packages/web/src/common/services/remote-config/remote-config.ts

```diff
@@ -218,6 +220,29 @@ export const remoteConfig = <
     await new Promise<void>(resolve => onClientReady(() => resolve()))
   }
 
+  /**
+   * Need this function for feature flags that depend on user id.
+   * This is because the waitForRemoteConfig does not ensure that
+   * user id is available before its promise resolution, meaning
+   * that it will sometimes return false for a feature flag
+   * that is enabled and supposed to return true.
+   */
+  const waitForUserRemoteConfig = async () => {
+    if (state.userId) {
+      await new Promise<void>(resolve => onClientReady(resolve))
```

Comment:
im not that familiar with promises, but can we just do `return onClientReady()` since async always returns a promise anyway?

Reply:
hmm, don't think that'll work because onClientReady needs the callback, which comes from the promise and lets us know we're done